Feature: add blog for index operation UI (#1365)
opensearch-project/project-website: _posts/2023-02-28-admin-panels-for-index-operations.md

---
layout: post
title:  "Index management UI enhancements"
authors:
- ihailong
- gbinlong
- suzhou
- lxuesong
date: 2023-02-28
categories:
 - technical-post
meta_keywords: index operations, split, shrink, reindex, index template, alias

> meta_description: Simplify cluster operations with index management UI enhancements.

Reviewer: Please update the meta keywords and description with the following: 

**Meta_keywords:**  index management, index mappings, index operation UI, shrink index operations, OpenSearch Dashboards
**Meta_description:** Simplify cluster operations with index management UI enhancements that enable you to open, close, reindex, shrink, and split indexes in a more user-friendly way.

Author: Done for this

### 5. Simplify the flow for the reindex operation.

You can easily select the source and destination indexes, aliases, or data streams from the dropdown menu. Additionally, we provide an ad-hoc destination index creation flow that allows you to import settings and mappings directly from the source, making the process more convenient.

![Image: Reindex page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/reindex.png){:.img-fluid }

### 6. Shrink index operation

Shrink index operation is used to shrink an existing index into a new index with fewer primary shards.

When you want to shrink an index, go to the indices pages, select one index, click the `Shrink` action in the actions menu then you will enter the shrink index page. Please notice that only one index can be shrunk at once and data stream indices are not supported to do shrink, so if multiple indices are selected or a data stream backing index is seletecd in the indices pages, the `Shrink` option is disabled. 

![Image: Shrink-action]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-action.jpg){:.img-fluid }

> Before shrinking, please make sure that the index you want to shrink is not in these states:

Reviewer 2: ```suggestion
Before shrinking, please make sure that the index you want to shrink is not in one of the following states:
```

> * the index's health status is red.

Reviewer 2: ```suggestion
* The index's health status is red.
```

> * the index has only one primary shard.

Reviewer 2: ```suggestion
* The index has only one primary shard.
```

> If the index is in one of the states above, some error messages will be shown in the shrink index page while the shrink button is disabled.

Reviewer 2: ```suggestion
If the index is in one of the states above, error messages will be shown in the shrink index page and the shrink button will be disabled.
```

![Image: Shrink-red]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-red.jpg){:.img-fluid }

![Image: Shrink-one-shard]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-one-shard.jpg){:.img-fluid }

Shrink index operation has some prerequisites, if the source index does not meet the conditions, some error messages will be shown in the shrink index page.

> One condition is that the source index must block write operations, i.e. the `index.blocks.write` setting in the source index is `true`, if the source index is not set to block write operations, then you can click the `Block write operations` button to set the `index.blocks.write` setting to `true`.

Reviewer 2: ```suggestion
The source index must block write operations. For example, if the `index.blocks.write` setting in the source index is set to `true`. If the source index is not set to block write operations, you can click the `Block write operations` button to set the `index.blocks.write` setting to `false`.
```

![Image: Shrink-block-write]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-block-write.jpg){:.img-fluid }

> The second condition is that the source index must be open, when the source index is closed, the shrink operation will fail, so you can click the `Open` button to open the index, this may take additional time to complete and the index will be in red health status while opening.

Reviewer 2: ```suggestion
Secondly, the source index must be open. When the source index is closed, the shrink operation will fail. You can click the `Open` button to open the index. This may take additional time to complete and the index will be in the red health status while opening.
```

![Image: Shrink-open]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-open.jpg){:.img-fluid }

> Another condition is that a copy of every shard in the source index must reside on the same node, you can move a copy of every shard to one node manually or using the shard allocation filter to do so, but it is not required when the cluster has only one data node or the replica count of the source index is equal to the number of data nodes minus 1(in these cases, a copy of every shard in the source index just reside on the same node). So if a copy of every shard in the source index does not reside on the same node, you can update the `index.routing.allocation.require._name` setting of the source index to move shards to one node automatically.

Reviewer 2: ```suggestion
Additionally, a copy of every shard in the source index must reside on the same node. You can move a copy of every shard to one node manually or use the shard allocation filter to do so, but this is not required when the cluster has only one data node or if the replica count of the source index is equal to the number of data nodes minus 1. In these cases, a copy of every shard in the source index reside on the same node. If a copy of every shard in the source index does not reside on the same node, you can update the `index.routing.allocation.require._name` setting of the source index to move shards to one node automatically.
```

You can update the setting in the source index's detail page:
![Image: Shrink-move-shards]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-move-shards.jpg){:.img-fluid }

> or submit the following request by `Dev Tools`:

Reviewer 2: ```suggestion
Or you can submit the following request with `Dev Tools`:
```

```
PUT test-1/_settings
{
"index.routing.allocation.require._name":"node1"
}
```

> After all things are ready, fill the input form in the shrink index page, speficy a name, primary shard count and replica count for the new shrunken index, you can also secify new aliases or select existing aliases to attach them to the new shrunken index. 

Reviewer 2: ```suggestion
After everything is ready, fill out the input form in the shrink index page, specify a name, the primary shard count, and the replica count for the new shrunken index. You can also specify new aliases or select existing aliases to attach to the new shrunken index. 
```

![Image: Shrink-configure]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-configure.jpg){:.img-fluid }

> All the settings you specified above will also be shown in the json editor contained in the `Advanced settings` part, you can specify more index settings for the new shrunken index in the json editor, for example, set both `index.routing.allocation.require._name` and `index.blocks.write` to `null`, this will clear the allocation requirement and the index write block copied from the source index.

Reviewer 2: ```suggestion
All of the settings specified above will also be shown in the json editor under the `Advanced settings` section. You can also specify additional index settings for the new shrunken index in the json editor. For example, set both `index.routing.allocation.require._name` and `index.blocks.write` to `null`. This will clear the allocation requirement and the index write block is copied from the source index.
```

![Image: Shrink-advanced-settings]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-advanced-settings.jpg){:.img-fluid }

> As the settings for the new shrunken index are specified, click the `Shrink` button then the shrink operation will be triggered:

Reviewer 2: ```suggestion
Now that the settings for the new shrunken index are specified, click the `Shrink` button. The shrink operation will be triggered:
```

![Image: Shrink-started]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-started.jpg){:.img-fluid }

> you can see a toast showing the shrink operation started successfully, after minutes or even hours, when the shrink operation is completed, a new toast will be shown says that the source index has been successfully shrunken.

Reviewer 2: ```suggestion
You will see a notification showing the shrink operation has started successfully. After the shrink operation is completed (this can be anywhere from minutes to hours), a new notification will be shown saying that the source index has been successfully shrunken.
```

![Image: Shrink-completed]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-completed.jpg){:.img-fluid }

### 7. Open and close index operation

> You can selecet multiple indices except the backing indices of a data stream to open or close.

Reviewer 2: ```suggestion
You can select multiple indices, except for the backing indices of a data stream, to open or close.
```

> If you don't need to read or search some old indices, but you don't want to delete them, then you can use close opeartion to     close these indices which can maintain the data but have a small overhead on the cluster. Another scenario is that when you want to add an new analyzer to an existing index, you must close the index, define the analyzer and then open the index. A closed index is blocked for read and write operations, so you must type the word close to confirm your action:

Reviewer 2: ```suggestion
If you don't need to read or search old indices, but you don't want to delete them, you can use close operation to     close indices which will maintain the data but will  then have small overhead on the cluster. Additionally, when you want to add a new analyzer to an existing index you must close the index, define the analyzer, then open the index. A closed index is blocked for read and write operations so you must type the word close to confirm your action:
```

![Image: Close-index]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/close-index.jpg){:.img-fluid }

> Open index is easier, you can select multiple indices to open even though these indices are all open:

Reviewer 2: ```suggestion
Opening an index is easier as you can select multiple indices to open even though some indices are already open:
```

![Image: Open-index]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/open-index.jpg){:.img-fluid }

### 8. Split index

> When user choose to split an index, we'll check the status of that index, if it's not ready to be splitted, we'll provide actions that user can operate to make the index ready for splitting. We'll also provide a list of shard numbers the index can be splitted to so user does not need to calculate it manually.

Reviewer 2: ```suggestion
When users choose to split an index the status of that index is checked. If the index not able to be split, actions the user can take to make the index ready for splitting are provided. A list of shard numbers the index can be split into are provided so that the user does not need to calculate it manually.
```

> User can  specify number of replicas, associate the new index with existing index alias or new alias. They can also use JSON editor to specify any index setting as well.

Reviewer 2: ```suggestion
Users can specify the number of replicas and then associate the new index with an existing index or use a new alias. They can also use the JSON editor to specify any index settings as well.
```

![Image: Split page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/splitindex-normal.jpg){:.img-fluid }
![Image: Split page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/splitindex-notready.jpg){:.img-fluid }

# In the end

**Try it**

> To try the new index management, please find features on [official playground](https://playground.opensearch.org/app/opensearch_index_management_dashboards#/indices).

Reviewer 2: ```suggestion
To try out the new index management features, check out these features on the [official playground](https://playground.opensearch.org/app/opensearch_index_management_dashboards#/indices).
```

## Next steps

> We’ll be releasing more Index Management UI features related to data streams, monitor metrics and so on. If you have any advice on what we currently have or will release, please leave a message on the [OpenSearch forum](https://forum.opensearch.org/).

Reviewer: Consider referencing OpenSearch 2.5 in here and linking back to the blog: https://opensearch.org/blog/opensearch-2-5-is-live/ since it mentions the index management enhancements.

Author: Done for this.

---
layout: post
title:  "Index Management UI enhancements"
authors:
- ihailong
- gbinlong
- suzhou
- lxuesong
date: 2023-02-28

> categories:

Reviewer: Please update blog date to March 1, 2023 if you plan to push it live today.

Author: Done for this

 - technical-post
meta_keywords: index management, index mappings, index operation UI, shrink index operations, OpenSearch Dashboards
meta_description: Simplify cluster operations with index management UI enhancements that enable you to open, close, reindex, shrink, and split indexes in a more user-friendly way.

excerpt: OpenSearch now offers users a new interface for running common indexing and data stream operations. Users can perform create, read, update, and delete (CRUD) and mapping for indexes, index templates, and aliases through the UI as well as open, close, reindex, shrink, and split indexes. This is the first step toward establishing an Index Management UI, which will serve as a unified administration panel in OpenSearch Dashboards.
---

> Users may find it difficult to manage indexes, aliases, and templates on Opensearch Dashboard using only APIs. There are restrictions and relation-binding tasks, like adding aliases to indexes, simulating an index template by its name and more. We are excited to announce that these cluster operations have now been largely simplified with index management UI enhancements on Opensearch Dashboard v2.5.

Reviewer 3: ```suggestion
Users may find it difficult to manage indexes, aliases, and templates in Opensearch Dashboards using only APIs because there are restrictions and relation-binding tasks, like adding aliases to indexes, simulating an index template by its name, and more. We are excited to announce that these cluster operations have now been largely simplified with index management UI enhancements in Opensearch Dashboard v2.5.
```

### Visual editor for index mappings with nested tree editor

> It is more complicated to build a JSON when it comes to index mappings, which have multiple nested layers and properties. To simplify that, a visual editor is provided with editable capabilities for nested properties. Users can add properties by clicking the operation buttons and then see what the mappings will look like by switching to the JSON editor.

Reviewer 3: ```suggestion
It is more complicated to build a JSON when it comes to index mappings, which have multiple nested layers and properties. To simplify this task, a visual editor is provided, with editable capabilities for nested properties. Users can add properties by clicking the operation buttons and then see what the mappings will look like by switching to the JSON editor.
```

Reviewer 3: The meaning of "It is more complicated to build a JSON when it comes to index mappings" is unclear. It is more complicated than what? A JSON what? Replace "when it comes to" with something simpler.

> <img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/index-mapping-editor.png" alt="index mapping editor"/>{: .img-fluid }

Reviewer 3: Introduce the image in the text that precedes it. Often, ", as shown in the following image" will suffice.

## Simulate an index by index name

> It is challenging to predict the appearance of the index when taking into account existing templates. To resolve this issue, we will attempt to match the index name with the templates whenever you change the index name, as shown in the image below. We will merge what users manually input with what the matching template contains, making it a "what you see is what you get" index.

Reviewer 3: ```suggestion
It is challenging to predict the appearance of the index when taking into account existing templates. To resolve this issue, we will attempt to match the index name with the templates whenever you change the index name, as shown in the following image. We will merge what users manually input with what the matching template contains, making it a "what you see is what you get" index.
```

> It is challenging to predict the appearance of the index when taking into account existing templates. To resolve this issue, we will attempt to match the index name with the templates whenever you change the index name, as shown in the image below. We will merge what users manually input with what the matching template contains, making it a "what you see is what you get" index.

Reviewer 3: Instead of "we", do we mean "OpenSearch will attempt" and "OpenSearch will merge what users..."? "We" is unclear in these instances, so it's better to name what we're referring to.

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/simulate-template.png" alt="simulate template"/>{: .img-fluid }

## Edit settings by JSON editor and diff mode

The new Index Management UI offers a visual editor and a JSON editor, which you can use if the visual editor does not support all available fields. Additionally, we provided the editor with a `diff` mode, as shown in the following image, so that you can compare the existing index and see what changes they have made to ensure that no mistakes are made.

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/json-diff-editor.png" alt="JSON diff editor"/>{: .img-fluid }

> ### One click to manage aliases on indexes

Reviewer 3: ```suggestion
### One click to manage index aliases
```

> It is straightforward to find out what aliases an index container through API, but it is difficult to determine how many indexes an alias points to through the API. The **Aliases** page provides you with these results by grouping indexes by alias. With the Index Management UI, it will be much effortless to attach or detach indexes from an alias, and the alias actions will be automatically generated, eliminating the need for you to build alias actions to add or remove indexes behind an alias and manually enter the indexes, which can lead to mistakes.

Author: > Use judgment-free language. Words like _simple_, _easy_, and _just_ create a skill judgment that may not apply to everyone in the OpenSearch community. Avoid Latinisms like "via." In the third sentence forward, will these things be the case in the future (we use future tense), or are they true now because of the enhancements? Reference the image in the text that precedes it.

Restore the previous comment and merge the suggestions.

> It is straightforward to find out what aliases an index container through API, but it is difficult to determine how many indexes an alias points to through the API. The **Aliases** page provides you with these results by grouping indexes by alias. With the Index Management UI, it will be much effortless to attach or detach indexes from an alias, and the alias actions will be automatically generated, eliminating the need for you to build alias actions to add or remove indexes behind an alias and manually enter the indexes, which can lead to mistakes.

Reviewer 3: ```suggestion
It is straightforward to find out what aliases an index contains through the API, but it is difficult to determine how many indexes an alias points to through the API. The **Aliases** page provides you with these results by grouping indexes by alias. With the Index Management UI, it will be easier to attach or detach indexes from an alias, and the alias actions will be automatically generated, eliminating the need for you to build alias actions to add or remove indexes behind an alias and manually enter the indexes, which can lead to mistakes.
```

Reviewer 3: We can leave "easier" in this case because there isn't a better alternative.

---
layout: post

> title:  "Index management UI enhancements"

Reviewer 3: ```suggestion
title:  "Index Management UI enhancements"
```

authors:
- ihailong
- gbinlong
- suzhou
- lxuesong
date: 2023-02-28
categories:
 - technical-post
meta_keywords: index management, index mappings, index operation UI, shrink index operations, OpenSearch Dashboards

> meta_description: Simplify cluster operations with index management UI enhancements that enable you to open, close, reindex, shrink, and split indexes in a more user-friendly way.

Reviewer 3: ```suggestion
meta_description: Simplify cluster operations with Index Management UI enhancements that enable you to open, close, reindex, shrink, and split indexes in a more user-friendly way.
```

### Visual editor for index mappings with nested tree editor

Building a JSON for index mappings is complicated because it has several nested layers and properties. To simplify this task, we now provide a visual editor, which is shown in the following image, with editable nested properties. You can add properties by clicking the operation buttons and then see what the mappings will look like by switching to the JSON editor.

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/index-mapping-editor.png" alt="index mapping editor"/>{: .img-fluid }

### Simulate index by index name

It is challenging to predict the appearance of the index taking into account existing templates. To resolve this issue, we will attempt to match the index name with the templates whenever you change the index name. We will merge what users manually input with what the matching template contains, making it a "what you see is what you get" index.

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/simulate-template.png" alt="simulate template"/>{: .img-fluid }

### Edit settings by JSON editor and diff mode

> The new Index Management UI offers a visual editor and a JSON editor, which you can use if the visual editor does not support all available fields. Additionally, we provided the editor with a `diff` mode, as shown in the following image, so that you can compare the existing index and see what changes they have made to ensure that no mistakes are made.

Author: How about "so that you can compare the existing index with the index in old version"

> The new Index Management UI offers a visual editor and a JSON editor, which you can use if the visual editor does not support all available fields. Additionally, we provided the editor with a `diff` mode, as shown in the following image, so that you can compare the existing index and see what changes they have made to ensure that no mistakes are made.

Reviewer 3: That works, but you need a "the" before "old": "so that you can compare the existing index with the index in the old version." Made the change in the suggestion.

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/json-diff-editor.png" alt="JSON diff editor"/>{: .img-fluid }

> ### One click to manage aliases on indexes

Author: Agree, "One click to manage index aliases"

<img src="/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/alias-creation.png" alt="create an alias"/>{: .img-fluid }

### Simplify the flow for the reindex operation

You can select the source and destination indexes, aliases, or data streams from the dropdown menu as seen in the following image. Moreover, we provide an ad-hoc destination index creation flow that allows you to import settings and mappings directly from the source.

![Image: Reindex page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/reindex.png){:.img-fluid }

### Shrink index operation

The **Shrink index** operation is used to reduce the number of primary shards in an existing index and create a new index.

To shrink an index, go to the **Indices** page, select an index, and choose the **Shrink** action in the **Actions** menu. This will take you to the shrink index page as seen in the following image.

![Image: Shrink-action]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-action.jpg){:.img-fluid }

> Only one index can be shrunk at once, and data stream indexes are not supported for a shrink operation. If multiple indexes are selected or a data stream backing an index is selected in the **Indices** page, the **Shrink** option is disabled.

Reviewer 3: ```suggestion
Only one index can be shrunk at once, and the shrink operation does not support data stream indexes. If multiple indexes are selected or if a data stream backing an index is selected in the **Indices** page, the **Shrink** option is disabled.
```

{: .note }

Before shrinking an index, ensure that the index is not in one of the following states:

* The index's health status is red.
* The index has only one primary shard.

If the index is in any of the above states, error messages will be displayed on the **Shrink index** page, and the **Shrink** button will be disabled as seen in the following images.

![Image: Shrink-red]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-red.jpg){:.img-fluid }

![Image: Shrink-one-shard]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-one-shard.jpg){:.img-fluid }

> The **Shrink index* operation requires certain prerequisites to be met. If the source index does not meet the specified conditions, error messages will be displayed on the **Shrink index** page.

Author: Should be a typo

The source index must block write operations, which means that you need to set the `index.blocks.write` setting in the source index to `true`. If the source index is not set to block write operations, you can choose the **Block write operations** button to set the `index.blocks.write` setting to `true` as seen in the following image.

![Image: Shrink-block-write]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-block-write.jpg){:.img-fluid }

> The source index must also be open. If the source index is closed, the shrink operation will fail. You can choose the **Open** button to open the index as seen in the following image. This may take additional time to complete, and the index will be in the red health status while opening.

Reviewer 3: ```suggestion
The source index must also be open. If the source index is closed, the shrink operation will fail. You can choose the **Open** button to open the index, as shown in the following image. This may take additional time to complete, and the index will be in the red health status while opening.
```

![Image: Shrink-open]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-open.jpg){:.img-fluid }

A copy of every shard in the source index must reside on the same node. You can move a copy of every shard to one node manually or use the shard allocation filter to do so. However, this is not required when the cluster has only one data node or if the replica count of the source index is equal to the number of data nodes minus one. In these cases, a copy of every shard in the source index resides on the same node. If a copy of every shard in the source index does not reside on the same node, you can update the `index.routing.allocation.require._name` setting of the source index to move shards to one node automatically.

> You can update this setting in the source index's detail page as seen in the following image.

Reviewer 3: ```suggestion
You can update this setting on the source index's detail page, as shown in the following image.
```

![Image: Shrink-move-shards]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-move-shards.jpg){:.img-fluid }

Or you can submit the following request with `Dev Tools`:

```
PUT test-1/_settings
{
"index.routing.allocation.require._name":"node1"
}
```

> After ensuring that all prerequisites are met, you can fill out the input form on the **Shrink index** page. Specify a name, the primary shard count, and the replica count for the new shrunken index as seen in the following image. You can also specify new aliases or select existing aliases to attach to the new shrunken index.

Reviewer 3: ```suggestion
After ensuring that all prerequisites are met, you can complete the input form on the **Shrink index** page. Specify a name, the primary shard count, and the replica count for the new shrunken index, as shown in the following image. You can also specify new aliases or select existing aliases to attach to the new shrunken index.
```

![Image: Shrink-configure]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-configure.jpg){:.img-fluid }

> All of the settings specified above will also be shown in the JSON editor under the **Advanced settings** section. You can also specify additional index settings for the new shrunken index in the JSON editor. For example, you can set both `index.routing.allocation.require._name` and `index.blocks.write` to `null` as seen in the following image. This will clear the allocation requirement and the index write block, which are copied from the source index.

Reviewer 3: ```suggestion
All of the specified settings will also be shown in the JSON editor under the **Advanced settings** section. You can also specify additional index settings for the new shrunken index in the JSON editor. For example, you can set both `index.routing.allocation.require._name` and `index.blocks.write` to `null`, as shown in the following image. This will clear the allocation requirement and the index write block, which are copied from the source index.
```

![Image: Shrink-advanced-settings]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-advanced-settings.jpg){:.img-fluid }

> Once you have specified all of the settings for the new shrunken index, choose the **Shrink** button to trigger the shrink operation. You will see a notification indicating that the shrink operation has started successfully as seen in the following image.

Reviewer 3: ```suggestion
Once you have specified all of the settings for the new shrunken index, choose the **Shrink** button to trigger the shrink operation. You will see a notification indicating that the shrink operation has started successfully, as shown in the following image.
```

![Image: Shrink-started]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-started.jpg){:.img-fluid }

> After the shrink operation is completed, which may take anywhere from minutes to hours, a new notification will be displayed stating that the source index has been successfully shrunken as seen in the following image.

Reviewer 3: ```suggestion
After the shrink operation is complete, which may take anywhere from minutes to hours, a new notification will be displayed stating that the source index has been successfully shrunk, as shown in the following image.
```

![Image: Shrink-completed]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/shrink-completed.jpg){:.img-fluid }

> ### Open and close index operation

Reviewer 3: ```suggestion
## Open and close index operations
```

You can select multiple indexes, except for the backing indexes of a data stream, to open or close.

> If you no longer need to read or search old indexes but do not want to delete them, you can use the **Close** operation to close indexes. This will maintain the data, but will have a small overhead on the cluster. Additionally, when you want to add a new analyzer to an existing index, you must close the index, define the analyzer, and then open the index. A closed index is blocked for read and write operations, so you must type the word "close" to confirm your action as seen in the following image.

Reviewer 3: ```suggestion
If you no longer need to read or search old indexes but do not want to delete them, you can use the **Close** operation to close indexes. This will maintain the data but will have a small overhead on the cluster. Additionally, when you want to add a new analyzer to an existing index, you must close the index, define the analyzer, and then open the index. A closed index is blocked for read and write operations, so you must enter the word "close" to confirm your action, as shown in the following image.
```

Reviewer 3: The meaning of "will have a small overhead" is unclear. Please rephrase to simplify/clarify.

Reviewer 3: I like it, but do we mean "This will maintain the data by occupying a small amount of overhead on the cluster"? If yes, I've already made the change in my suggestion.

![Image: Close-index]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/close-index.jpg){:.img-fluid }

> Opening an index is easier as you can select multiple indexes to open even if some indexes are already open as seen in the following image.

Reviewer 3: ```suggestion
You can select multiple indexes to open, even if some indexes are already open, as shown in the following image.
```

![Image: Open-index]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/open-index.jpg){:.img-fluid }

> ### Split index

Reviewer 3: ```suggestion
## Split index
```

Reviewer 3: Should this be "Split index operation"?

> When you choose to split an index, the status of that index is checked. If the index is not able to be split, actions that the user can take to make the index ready for splitting are provided. A list of shard numbers that the index can be split into is also provided so that the user does not need to calculate it manually.

Reviewer 3: ```suggestion
When you choose to split an index, the status of that index is checked. If the index is not able to be split, actions that you can take to prepare the index for splitting are provided. A list of shard numbers that the index can be split into is also provided so that you do not need to calculate it manually.
```

Reviewer 3: Last sentence: Instead of "it", let's name what we're referring to. "the numbers"?

Author: Yes, "the shard numbers" .

> You can then specify the number of replicas and associate the new index with an existing index or use a new alias. You can also use the JSON editor to specify any additional index settings as seen in the following images.

Reviewer 3: ```suggestion
You can then specify the number of replicas and associate the new index with an existing index or use a new alias. You can also use the JSON editor to specify any additional index settings, as shown in the following images.
```

![Image: Split page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/splitindex-normal.jpg){:.img-fluid }
![Image: Split page]({{site.baseurl}}/assets/media/blog-images/2023-02-28-admin-panels-for-index-operations/splitindex-notready.jpg){:.img-fluid }

> ## In conclusion

Reviewer 3: ```suggestion

```

> **Try it**

Reviewer 3: ```suggestion

```

> If you want to try out the new index management features, you can check them out on the [official OpenSearch playground](https://playground.opensearch.org/app/opensearch_index_management_dashboards#/indices).

Reviewer 3: ```suggestion

```

## Next steps

> If you want to try out the new index management features, you can check them out on the official OpenSearch playground. We will also be releasing more Index Management UI features for data streams, monitor metrics, and more. If you have any feedback or suggestions on what we currently have or will release, please leave a message on the [OpenSearch forum](https://forum.opensearch.org/).

Reviewer 3: ```suggestion
If you want to try the new index management features, you can experiment with them on the [OpenSearch playground](https://playground.opensearch.org/app/opensearch_index_management_dashboards#/indices). We will also be releasing more Index Management UI features for data streams, metrics monitoring, and more. If you have any feedback or suggestions, leave a message on the [OpenSearch forum](https://forum.opensearch.org/).
```

## References

> We are excited to announce that [OpenSearch 2.5](https://opensearch.org/blog/opensearch-2-5-is-live/) is now live!

Reviewer 3: ```suggestion
* [OpenSearch 2.5 is live!](https://opensearch.org/blog/opensearch-2-5-is-live/) blog post
```
